### PR TITLE
Preserve newlines before comments

### LINKED ISSUE #62: Newline after function-like not preserved
The formatter currently does not preserve newlines after function-like decls (`event`, `function`, `hook`), e.g.,

```zeek
event zeek_init() {}
# event zeek_init() {}
```

is formatted as
```zeek
event zeek_init()
	{ }# event zeek_init() {}
```

Formatting the output again leads to more reformatting (new space after `#`) pointing to an idempotency violation (#26),
```zeek
event zeek_init()
	{ } # event zeek_init() {}
```

This makes it hard in code where one needs to selectively comment out blocks of code.

### PR DESCRIPTION
**Pull request by bbannier**
Closes #62.

### CODE REVIEW

File: tests/test_formatting.py [hunk 1]
```diff
@@ -62,6 +63,24 @@ def test_index_slice(self):
...
+        regression test for #62."""
+        code = textwrap.dedent(
+            """\
+               event zeek_init()
+               \t{ }
+               # Comment on next line.
+
+               global x = 42; # Comment on same line.
+            """
+        ).encode()
```
Comment (awelzel) [inline tests/test_formatting.py:77]: Unless I'm doing something completely wrong, I can't reproduce #62 with the latest zeek-script release (or when running from the `main` branch). The test seems useful, but it also passes if I don't pick up the MinorCommentFormatter change.

```
$ docker run --entrypoint=/bin/bash --rm -it -v $(pwd):/src -w /src python:3
# pip install zeekscript
# cat nl-preserve.zeek 
event zeek_init()
        {}
# Comment on next line.

global x = 42; # Comment on same line.
# zeek-script format < nl-preserve.zeek 
event zeek_init()
        { }
# Comment on next line.

global x = 42; # Comment on same line.
```

**Review by awelzel — COMMENTED** [review]
Not quite following what/why it's broken.

File: tests/test_formatting.py (hunk 1 above)
Comment (bbannier) [reply tests/test_formatting.py]: Looks like the previous version works if we drop the line with the `global` which I added later. I broke that out into a separate check now.

Comment (awelzel) [reply tests/test_formatting.py]: Now I'm even more confused, but approving as I hope it makes sense to you and does obviously fix something you run into.

**Review by awelzel — APPROVED** [review]

File: tests/test_formatting.py (hunk 1 above)
Comment (bbannier) [reply tests/test_formatting.py]: I agree, this is super confusing and I also do not have a good handle on local reasoning here. At least we have a smallish test case now so this exact issue cannot happen anymore 😞.

**Pull request merged**